packages/vike-react-styled-jsx/Wrapper.tsx

```diff
@@ -0,0 +1,21 @@
+export { Wrapper }
+
+import React, { type ReactNode } from 'react'
+import { StyleRegistry, createStyleRegistry } from 'styled-jsx'
+import { usePageContext } from 'vike-react/usePageContext'
+
+function Wrapper({ children }: { children: ReactNode }) {
+  const pageContext = usePageContext()
+
+  if (isBrowser() || pageContext.config.styledJsx === null) return <>{children}</>
+
+  pageContext.config.styledJsx ??= createStyleRegistry()
```

Comment:
I ain't sure I follow the logic: do we want to allow the user to instantiate `createStyleRegistry()`?

In principle, we should never mutate `pageContext.config`. The `pageContext.config` object is the source of truth holding user configuration. Destroying source of truth is bad.

Reply:
> do we want to allow the user to instantiate `createStyleRegistry()`?

No, since `createRegistry()` has no props. If `+styledJsx === null` then we just return `children`. 
It's okay if we just do `pageContext.config.styledJsx = createStyleRegistry()` (without `??`) ?

Reply:
Why should we go for a hacky solution if we don’t need to? See my comment about source of truth.

Reply:
I've improved the code and add support for adding `csp nonce`. Please take a look : https://github.com/vikejs/vike-react/pull/158/commits/6f2a2d8fba6d36c198f0c01675b11d8db8916122.
Do you think it's fine to do `pageContext.styledJsxRegistry = createStyledRegistry()` ?, since it's not passed to client, I think it's okay cause we don't need to serialize it.

If you think it's better to do it inside `+onBeforeRenderHtml`, then I'll move it.

Reply:
> Do you think it's fine to do `pageContext.styledJsxRegistry = createStyledRegistry()` ?, since it's not passed to client, I think it's okay cause we don't need to serialize it.

Yes, that's much cleaner. It's a good practice to avoid destroying source of truths.

Reply:
> If you think it's better to do it inside `+onBeforeRenderHtml`, then I'll move it.

Yea, maybe. I feel like using `+onBeforeRenderHtml` would be more explicit.